snuba/web/query.py

```diff
@@ -165,8 +167,36 @@ def _format_storage_query_and_run(
     """
 
     source = clickhouse_query.get_data_source().format_from()
-    with sentry_sdk.start_span(description="create_query", op="db"):
-        formatted_query = DictSqlQuery(dataset, clickhouse_query, request_settings)
+    with sentry_sdk.start_span(description="create_query", op="db") as span:
+        legacy_query = DictSqlQuery(dataset, clickhouse_query, request_settings)
+        span.set_data("legacy_query", legacy_query.sql_data())
+        try:
+            ast_query: Optional[AstSqlQuery] = AstSqlQuery(
+                clickhouse_query, request_settings
+            )
+            # For mypy
+            assert ast_query is not None
+            span.set_data("ast_query", ast_query.sql_data())
+
+        except Exception:
+            logger.warning("Failed to format ast query", exc_info=True)
```

Comment:
Any reason this shouldn't be logged as an error instead of just warning? What does exc_info=True mean again?

Reply:
It is not logged as error (yet) because, as long as the system is not rolled out this does not fail the query.
I would move directly from warning to throwing when we are rolled out.

Reply:
Ok. You mean after we are 100% rolled out? Since there's no code change associated with rolling out, we'll be logging warnings still when we start using the AST right?

Reply:
I would actually start enforcing this throws before we are at 100% when the formatter is stable enough. But yeah that would be the plan.